src/v/rpc/connection_cache.cc

```diff
@@ -20,7 +20,18 @@ namespace rpc {
 connection_cache::connection_cache(
   ss::sharded<ss::abort_source>& as,
   std::optional<connection_cache_label> label)
-  : _label(std::move(label)) {
+  : _max_connections(8)
+  , _label(std::move(label)) {
+    _as_subscription = as.local().subscribe(
+      [this]() mutable noexcept { shutdown(); });
+}
+
+connection_cache::connection_cache(
+  config::binding<size_t> max_connections,
```

Comment:
What is the point of passing it through as a binding? The value is used once and saved in a member so it could just be a `size_t` which would simplify things.

Reply:
Good catch, I wasn't sure at first if this is something we'd want to make dynamically configurable without a restart. But I don't believe we do. So I'll definitely just switch this over to `size_t`